src/c/redundancy/rasta_red_multiplexer.c

```diff
@@ -238,22 +234,17 @@ void redundancy_mux_close(redundancy_mux *mux) {
 
     mux->port_count = 0;
     rfree(mux->transport_sockets);
-    for (unsigned int i = 0; i < mux->redundancy_channels_count; i++) {
-        redundancy_channel_free(&mux->redundancy_channels[i]);
-    }
-    rfree(mux->redundancy_channels);
+    redundancy_channel_free(mux->redundancy_channel);
+    rfree(mux->redundancy_channel);
     rfree(mux->listen_ports);
 
     freeRastaByteArray(&mux->sr_hashing_context.key);
 }
 
 rasta_redundancy_channel *redundancy_mux_get_channel(redundancy_mux *mux, unsigned long id) {
```

Comment:
If we only have one redundancy channel, this method becomes a bit pointless. Also, it is called in only in `receive_packet()` (and in `redundancy_mux_wait_for_entity()`, which we want to remove as soon as possible, but it's still used by one test). Therefore, we could maybe move this check into the caller.